net/yggdrasil/files/yggdrasil.init

```diff
@@ -12,7 +12,7 @@ start_service()
 	procd_open_instance
 	procd_set_param respawn
 
-	procd_set_param command /usr/sbin/yggdrasil -useconffile /etc/yggdrasil.conf
+	procd_set_param command /bin/ash -c "ygguci get | yggdrasil -useconf"
```

Comment:
This seems correct. OpenWrt procd should always restart with latest UCI settings.  

The code below _does not_  re-write the `/etc/config/yggdrasil` file if yggdrasil is killed or crashes. The `/etc/init.d/yggdrasil` function `start_service()`  is not called after a crash or `killall yggdrasil`. Only `procd_set_param` values are re-executed.

```
start_service()
{
  procd_open_instance
  procd_set_param respawn

  ygguci get > /tmp/etc/yggdrasil.conf
  procd_set_param command /usr/sbin/yggdrasil -useconffile /tmp/etc/yggdrasil.conf
....
```

Is there a way we can keep `procd_set_param` to a single command, but always regenerate configs from uci on crash or kill? Do we keep the example code because it re-writes the config (as intended) with `/etc/init.d/yggdrasil restart`?

Reply:
@neheb could you please advise how to handle our procd launching yggdrasil? We had changed procd launching yggdrasil this way because we simply had /etc/yggdrasil.conf. Currently, we're looking for help to launch with the new `/usr/bin/ygguci` handler code.  

Ideas?

Reply:
@feckert -- any ideas? I'm thinking this is/was an issue previously but now I don't have a solution going forward?

Reply:
Why do we need to recreate the `/tmp/etc/yggdrasil.conf` file, if the process dies and the service is restarted by `procd`. There are no changes in the uci config?

Reply:
ping @wfleurant

Reply:
We can also add -normaliseconf call after ygguci get. 

```
/bin/ash -c "ygguci get | yggdrasil -useconf -normaliseconf -json | yggdrasil -useconf"
```

This way it will always run with correct config even if newer versions of yggdrasil will introduce some radical config changes. What do you think?

Reply:
I am not familar with you tool.
If we need this then i am fine.
My question was why do we need to recreate the /tmp/etc/yggdrasil.conf from uci if we get prespawned by procd?
I thought that was the intention by @wfleurant ?

Reply:
No, you're right. There is no need to re/create `/tmp/etc/yggdrasil.conf`  

The reason it is created stems from a feature in luci-app-cjdns. It allows the manual editing of config from LuCI. We're not there yet re: `luci-app-yggdrasil`

Reply:
@wfleurant good to have you back! I'm testing the luci app already, check your DMs in Matrix. Changing settings via LuCI works just fine.

What do you think about -normaliseconf, should I commit this change? I think it could be useful, they do that trick in Debian packages.

Reply:
Yes, you're right. It's best to use -normaliseconf -- re: Matrix.
> yup, it sets every setting possible. one thing may break it, if a user will set something stupid manually without a web interface
> The -normaliseconf Yggdrasil option lets you sanitise the config to avoid this kind of problem